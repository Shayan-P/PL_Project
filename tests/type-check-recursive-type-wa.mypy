checked

def f() -> int:
    x = f();
    print(x[3]);
    return x;
;