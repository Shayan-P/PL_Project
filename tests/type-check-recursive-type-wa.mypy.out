Type Error: type error in function call
inner trace: 

(app-expr (ident-expr '$index) (ident-expr 'x))


(app-expr (app-expr (ident-expr '$index) (ident-expr 'x)) (num-expr 3))


(app-expr
 (ident-expr 'print)
 (app-expr (app-expr (ident-expr '$index) (ident-expr 'x)) (num-expr 3)))


(app-expr
 (app-expr
  (ident-expr 'print)
  (app-expr (app-expr (ident-expr '$index) (ident-expr 'x)) (num-expr 3)))
 (end-of-args-expr))


(side-effect-stmt
 (app-expr
  (app-expr
   (ident-expr 'print)
   (app-expr (app-expr (ident-expr '$index) (ident-expr 'x)) (num-expr 3)))
  (end-of-args-expr)))


(def-stmt
 'f
 '()
 (primitive-type (int-type))
 (list
  (assign-stmt 'x (unknown-type) (app-expr (ident-expr 'f) (end-of-args-expr)))
  (side-effect-stmt
   (app-expr
    (app-expr
     (ident-expr 'print)
     (app-expr (app-expr (ident-expr '$index) (ident-expr 'x)) (num-expr 3)))
    (end-of-args-expr)))
  (return-value-stmt (ident-expr 'x))))

